nimare/meta/kernel.py

```diff
@@ -262,19 +269,18 @@ def _transform(self, mask, coordinates):
         dims = mask.shape
         vox_dims = mask.header.get_zooms()
 
-        transformed = []
-        for id_, data in coordinates.groupby("id"):
-            ijks = np.vstack((data.i.values, data.j.values, data.k.values)).T
-            kernel_data = compute_kda_ma(
-                dims, vox_dims, ijks, r=self.r, value=self.value, sum_overlap=False
-            )
-            transformed.append((kernel_data, id_))
-        return transformed
+        ijks = coordinates[["i", "j", "k"]].values
+        exp_idx = coordinates["id"].values
+        transformed = compute_kda_ma(
+            dims, vox_dims, ijks, self.r, self.value, exp_idx, sum_overlap=self._sum_overlap
+        )
+        exp_ids = np.unique(exp_idx)
+        return transformed, exp_ids
 
 
-class KDAKernel(KernelTransformer):
+class MKDAKernel(KDAKernel):
```

Comment:
some of the tests fail becuase they check if the kernel is an instance of KDAKernel, which MKDAKernel is now, [I pushed a commit to my fork fixing this](https://github.com/neurostuff/NiMARE/compare/performance-improvements...jdkent:performance-improvements)

Reply:
Thanks!